src/Core/ConstraintValidator/DefaultLanguageValidator.php

```diff
@@ -58,17 +52,33 @@ public function validate($value, Constraint $constraint)
             throw new UnexpectedTypeException($constraint, DefaultLanguage::class);
         }
 
+        if ($constraint->allowNull) {
+            if (null === $value) {
+                return;
+            }
+        }
+
         if (!is_array($value)) {
             throw new UnexpectedTypeException($value, 'array');
         }
 
-        if (empty($value[$this->defaultLanguageId])) {
+        if ($constraint->allowNull && !isset($value[$this->defaultLanguageContext->getId()]) && !isset($value[$this->defaultLanguageContext->getLocale()])) {
+            // Check that the array actually contains a value for the default language (regardless of its empty value or not), if no index
+            // matching the default language is set it means there is no planned modification
+            return;
+        }
+
+        // Check if value for default language is present, we use language ID for back-office form's usage, and we use the locale
+        // for Admin API usages since the input data is indexed by locale thanks to the LocalizedValue attribute
+        if (empty($value[$this->defaultLanguageContext->getId()]) && empty($value[$this->defaultLanguageContext->getLocale()])) {
```

Comment:
(NIT) `empty()` method can have really bad side-effects (see https://dev.to/klnjmm/never-use-empty-function-in-php-4pb0)

Reply:
Interesting read, that's when strict typing helps to avoid side effects

And in this case the type is absent on `$value` but we do check its type with `is_array` so we're sure of its type